src/authentication/UserAuthenticator.js

```diff
@@ -71,6 +71,11 @@ export class UserAuthenticator {
       if (userJson && userJson.error) {
         throw new Error(userJson.error);
       }
+
+      if (userJson === undefined) {
```

Comment:
Sorry, I clicked too early while typing a response 😅 .
I was thinking rather than checking specifically for undefined maybe it would be better to just split out the !userJson and userJson.UserID check?

```suggestion
      if (!userJson) {
        throw new Error(`userJson not found. Server Url ${serverURL}`);
      }
      
      if (!userJson.UserID) {
        throw new Error('Unexpected response from server');
      }
```

Although I'm surprised it doesn't just log a connection failure (I haven't reproduced the specific changing IP scenario). Maybe logging the response status code might also help with troubleshooting somewhere around here?
https://github.com/openmsupply/mobile/blob/19dc0094a4ca231fee8a54f5031f538b82d85a91/src/authentication/Helpers.ts#L61
```
logger.info(`Response status code: ${response?.status}`);
```

Reply:
@katherine-sussol : I think we need to handle both cases separately as the wrong URL gives undefined but if we have the correct URL and no name exists in mSupply then it might give us blank userJSON.